### PR TITLE
chore: Inject `COLLECTOR_VERSION` into Konflux-built images

### PR DESCRIPTION
**Pull request by msugakov**
## Description

Following up on ROX-19958, I decided to check if `make tag` for collector would print the same thing as we tag the image. It will not. Here I try to address that.

The only place I found where `COLLECTOR_TAG`/`COLLECTOR_VERSION` is meaningfully used is logging during the startup https://github.com/stackrox/collector/blob/8198c305e6d8ebaaeb31f44b54e2ec0cee5f2612/collector/container/scripts/bootstrap.sh#L78

This is, for reference, how `COLLECTOR_VERSION` is set downstream: https://gitlab.cee.redhat.com/stackrox/rhacs-midstream/-/blob/50310ae51698a9f1a27382764dd9957fb12ab75c/distgit/containers/rhacs-collector/Dockerfile.in#L109

I conclude, the precision is not very important there, it is free-form, and `3.18.x-284-ge3f5c5566c-fast` should be fine.

Also, as Tom looked into Konflux releases, I think `version` label contents aren't important for releasing (unlike downstream where it is important). Therefore, I'm addressing that thing.

If/when we will retag images, we can revisit the label but that's for later.

`NPROC` change is not connected to everything else but why not.

## Checklist
- [x] Investigated and inspected CI test results
- ~~[ ] Updated documentation accordingly~~ - not needed.

**Automated testing**

None will be added.

  - [ ] Added unit tests
  - [ ] Added integration tests
  - [ ] Added regression tests

## Testing Performed

- [x] CI is green enough (excluding Konflux Enterprise Contract which is ok to yell at us).
- [x] Konflux-built image still has `-fast` suffix in tag.
- [x] `make tag` manually ran in the repo does not fail and does not output `-fast`.
- [x] Konflux-built image has the same thing as tag in the `version` label and has `$COLLECTOR_VERSION` environment variable defined.
- [x] GHA-built image is still fine.

### CODE REVIEW

File: collector/container/konflux.Dockerfile [hunk 1]
```diff
@@ -133,12 +132,13 @@ LABEL \
     source-location="https://github.com/stackrox/collector" \
     summary="Runtime data collection for Red Hat Advanced Cluster Security for Kubernetes" \
     url="https://catalog.redhat.com/software/container-stacks/detail/60eefc88ee05ae7c5b8f041c" \
-    version=${COLLECTOR_VERSION} \
+    version="$COLLECTOR_TAG" \
```
Comment (tommartensen) [inline collector/container/konflux.Dockerfile:135]: [nit] 
```suggestion
    version="${COLLECTOR_TAG}" \
```

File: collector/container/konflux.Dockerfile [hunk 2]
```diff
@@ -133,12 +132,13 @@ LABEL \
...
     vendor="Red Hat, Inc."
 
 ARG BUILD_DIR
 ARG CMAKE_BUILD_DIR
 
+ENV COLLECTOR_VERSION="$COLLECTOR_TAG"
```
Comment (tommartensen) [inline collector/container/konflux.Dockerfile:141]: [nit]
```suggestion
ENV COLLECTOR_VERSION="${COLLECTOR_TAG}"
```

**Review by tommartensen — APPROVED** [review]
Two style nits, LGTM.

**Comment by tommartensen** [comment]
> Konflux-built image has the same thing as tag in the version label and has $COLLECTOR_VERSION environment variable defined.

Also the correct version is logged: 

```
$ podman run quay.io/rhacs-eng/collector:3.18.x-283-geb4d1224d4-fast
Collector Version: 3.18.x-283-geb4d1224d4-fast
OS: Linux
Kernel Version: 6.7.5-200.fc39.aarch64
Starting StackRox Collector...
...
```

**Comment by msugakov** [comment]
Regarding validation of GHA images.

Due to a [known thing](https://redhat-internal.slack.com/archives/CFMQ5C2TT/p1717172236203029?thread_ts=1717170093.796339&cid=CFMQ5C2TT), collector images in PR get a different tag.
For instance, in Konflux I get `3.18.x-284-gb177db993f` ([CI](https://console.redhat.com/preview/application-pipeline/workspaces/rh-acs/applications/acs/pipelineruns/collector-build-m5gbw)), same as when I run `make tag` locally. In GHA it is `3.18.x-285-g33f14d7e5e` ([CI](https://github.com/stackrox/collector/actions/runs/9321036911/job/25659668413?pr=1694#step:5:8)).
Apart from that, `COLLECTOR_VERSION=3.18.x-285-g33f14d7e5e` is there and `version` label is unset. Most important, we did not overwrite GHA images because `determine-image-tag` returned `-fast` in the tag `3.18.x-284-gb177db993f-fast`, and there is `quay.io/rhacs-eng/collector:3.18.x-284-gb177db993f-fast` but no `quay.io/rhacs-eng/collector:3.18.x-284-gb177db993f`.

**Review by Stringy — APPROVED** [review]

**Comment by msugakov** [comment]
Creation of test VMs is known to be broken ATM. https://redhat-internal.slack.com/archives/CFMQ5C2TT/p1717585747118389

Will merge.

**Pull request merged**